Apps/WebClient/src/ClientApp/src/components/calendar/body.vue

```diff
@@ -175,17 +175,29 @@ export default class CalendarBodyComponent extends Vue {
     @Prop() private isVisible!: boolean;
 
     private eventLimit: number = 4;
-    private firstEventDateOfCurrentMonth: Date | null = null;
 
     private isHovering: boolean = false;
     private hoveringEvent: CalendarEntry | null = null;
     private eventBus = EventBus;
-    private get currentDates() {
-        return this.getCalendar();
+
+    private currentDates: CalendarWeek[] = [];
+
+    @Watch("currentDate")
+    private onCurrentDateUpdate() {
+        this.currentDates = this.generateCalendar();
```

Comment:
naming? what are the currentDates? are they available dates or dates on which we have entries? We may need to come up with some meaningful names or adding some comments to make them clearer?

Reply:
I would read it as the currentDates for the calendar. From the type generateCalendar returns, I would infer that is an Array of Calendar weeks. Do you have a suggestion for the name?

Reply:
as I understand that the generateCalendar base on the date which is called currentDate, it returns all the dates on which there are events/entries. My suggestions are:
1. Passing the this.currentDate and this.slotEvents and whatever the function needs to base on as input params instead of reading it inside this method (per the PURE principle of Functional Programming which allows function to be unit-tested: i.e. "Given a parameter A → expect the function to return value B")
2. then we may call it generateMonthCalendar(monthDate, monthEvents) or shorter version is getMonthCalendar(...)
How do you think?

Reply:
I liked the passing of the parameter, to generate the calendar month. Regarding the dateGroups, I left as is since is part of the class members. We can refactor it once we optimize the generation of the calendar.
I've also renamed the method to your second suggestion, sounded much better!

Reply:
Sounds great! we try to get some opportunities to get our code a bit cleaner, as I don't think these are blocking us for now.